Source/Core/Core/State.cpp

```diff
@@ -434,6 +420,39 @@ static void CompressAndDumpState(CompressAndDumpState_args save_args)
     f.WriteBytes(buffer_data, buffer_size);
   }
 
+  const std::string last_state_filename = File::GetUserPath(D_STATESAVES_IDX) + "lastState.sav";
+  const std::string last_state_dtmname = last_state_filename + ".dtm";
+  const std::string dtmname = filename + ".dtm";
+
+  {
+    std::lock_guard lk(s_save_thread_mutex);
+
+    // Backup existing state (overwriting an existing backup, if any).
+    if (File::Exists(filename))
+    {
+      if (File::Exists(last_state_filename))
+        File::Delete((last_state_filename));
+      if (File::Exists(last_state_dtmname))
+        File::Delete((last_state_dtmname));
+
+      if (!File::Rename(filename, last_state_filename))
+        Core::DisplayMessage("Failed to move previous state to state undo backup", 1000);
+      else if (File::Exists(dtmname))
+        File::Rename(dtmname, last_state_dtmname);
```

Comment:
Should this have the same check as the `last_state_filename`?

Reply:
I guess, might as well.